File: 0003/LargestPrimeFactor.java
public class LargestPrimeFactor{
	
	public static void main(String args[]){

		long TO_FACTOR = 600851475143l;
		long present_factor = 2l;
		boolean factoring_done = false;
		long highest_prime = -1l;

		while(!factoring_done){
			if (TO_FACTOR % present_factor == 0l){
				TO_FACTOR /= present_factor;
				highest_prime = present_factor;
			} else {
				present_factor += 1l;
			}

			if (present_factor > TO_FACTOR){
				factoring_done = true;
			}
		}

		System.out.println(highest_prime);
		System.out.println(TO_FACTOR);

	}
}

File: 0003/largest_prime_factor.py
import math

PROMPT = 600851475143

def do_factor(to_factor):
	to_return = [to_factor]

	for n in range(2, int(math.pow(to_factor, 0.5))):
		if (to_factor % n == 0):
			to_return.remove(to_factor)
			to_return.extend(do_factor(to_factor / n))
			to_return.extend(do_factor(n))
			break

	return to_return

def main():
	largest_prime_factor = 1
	factor_list = do_factor(PROMPT)
		
	factor_list.sort()
	factor_list.reverse()

	print factor_list[0]

main()
